### PR TITLE
DEMO: Example of index/constraint on CourseClients, associations

### PR DESCRIPTION
**Pull request by francisli**

### CODE REVIEW

File: server/migrations/20240224200703-create-course-client.js [hunk 1]
```diff
@@ -27,15 +27,16 @@ module.exports = {
...
+        type: Sequelize.DATE,
       },
       updatedAt: {
         allowNull: false,
-        type: Sequelize.DATE
-      }
+        type: Sequelize.DATE,
+      },
     });
+    await queryInterface.createIndex('CourseClients', ['CourseId', 'ClientId'], { unique: true });
```
Comment (francisli) [inline server/migrations/20240224200703-create-course-client.js:37]: This "unique index" will prevent rows with the same CourseId, ClientId combination

File: server/models/course.js [hunk 2]
```diff
@@ -9,13 +9,18 @@ export default function (sequelize, DataTypes) {
      */
     static associate(models) {
       // define association here
+      Course.hasMany(models.CourseClient);
+      Course.belongsToMany(models.Client, { through: models.CourseClient });
```
Comment (francisli) [inline server/models/course.js:13]: Association examples

File: server/models/device.js [hunk 3]
```diff
@@ -7,29 +7,31 @@ export default function (sequelize, DataTypes) {
      * The `models/index` file will call this method automatically.
      */
     static associate(models) {
-      // define association here
+      Device.belongsTo(models.Donor);
```
Comment (francisli) [inline server/models/device.js:10]: Association examples

File: server/models/donor.js [hunk 4]
```diff
@@ -9,20 +9,24 @@ export default function (sequelize, DataTypes){
      */
     static associate(models) {
       // define association here
+      Donor.hasMany(models.Device);
```
Comment (francisli) [inline server/models/donor.js:12]: Association examples

File: server/migrations/20240224200703-create-course-client.js (hunk 1 above)
Comment (duanhash) [reply server/migrations/20240224200703-create-course-client.js:37]: For other people who use this as reference, the command is queryInterface.addIndex and not queryInterface.createIndex

Comment (francisli) [reply server/migrations/20240224200703-create-course-client.js:37]: Oops! Sorry about that...

**Pull request closed without merging**